### PR TITLE
:bfix: Fix CGAN backward issue

### PR DESCRIPTION
**Pull request by praeclarumjj3**
This PR fixes Issue #38. Replacing the problematic lines with the following ones works fine:

```python
# update
self.optim.zero_grad()
gen_loss.backward()
utils.average_gradients(self.model)
self.optim.step()

self.optimD.zero_grad()
dis_loss.backward()
utils.average_gradients(self.netD)
self.optimD.step()
```

The issue might have been due to the incorrect order of calculating the gradients and updating the weights. Doing the backward pass for Generator first works well!

### CODE REVIEW

**Comment by XiaohangZhan** [comment]
Thank you. But have you verify the testing results? Does this modification affect the final results?

**Comment by praeclarumjj3** [comment]
> Thank you. But have you verify the testing results? Does this modification affect the final results?

I trained the model for 45000 iterations with this modification implemented. It doesn't affect the result negatively, in my opinion.

You can see the results yourself here: https://github.com/praeclarumjj3/PCNet-C-Experiments#using-my-trained-model

For how many iterations did you train your model?

**Comment by XiaohangZhan** [comment]
Great! Thanks.
